<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <script>
      function throwErrr() {
        throw new Error("SOmethign went wron");
      }
      function handleErr() {
        try {
          const ans = throwErrr();
          console.log(ans);
        } catch (error) {
          console.log("errpr occuref", error);
        } finally {
          console.log(
            "Finally executrs thre code to be run regardless of error"
          );
        }
      }

      function division(a, b) {
        if (b == 0) {
          throw new Error("Denominator is zero");
        }
        return a / b;
      }
      try {
        const divi = division(5, 0);
        console.log(divi);
      } catch (error) {
        console.error("Error in dividing ", error);
      }

      const promiseErr = function () {
        return new Promise ((resolve,reject) => {
            setTimeout(() => {
                const success = false
               if(success) {
                resolve("Promise resolved")
               
               } else {
              reject("Promise rejected")
             
               }
            }, 2000);
        })
      }

      promiseErr().then((mssg) => console.log(mssg))
      .catch((err) => console.log(err))

      async function fetchData () {
try {
    
            const response = await fetch("https://jsonplaceholder.typicode.com/posts/2")
            if(!response) {
               throw new Error(response.status)
            }
            const data = response.json()
            console.log(data)
} catch (error) {
    console.log("Error",Error)
}
        
      }

      fetchData()
    </script>
  </body>
</html>
